plugin-server/src/utils/db/db.ts

```diff
@@ -214,11 +231,18 @@ export class DB {
             // Annotate query string to give context when looking at DB logs
             queryString = `/* plugin-server:${tag} */ ${queryString}`
             try {
+                let queryRes
                 if (client) {
-                    return await client.query(queryString, values)
+                    queryRes = await client.query(queryString, values)
                 } else {
-                    return await this.postgres.query(queryString, values)
+                    queryRes = await this.postgres.query(queryString, values)
+                }
+
+                if (cacheResult) {
+                    const queryResultToCache = { rows: queryRes.rows, rowCount: queryRes.rowCount }
+                    await this.redisSet(cacheKey, JSON.stringify(queryResultToCache), cacheResultTtlSeconds)
```

Comment:
This implementation doesn't avoid thundering herd to the database and redis when the key is not present and multiple workers are trying to cache it at the same time

Reply:
👍  will look into a good way of doing this

Reply:
probably can use a lot of the existing redlock logic we have

Reply:
addressed this